airbyte-cdk/python/airbyte_cdk/sources/file_based/availability_strategy/default_file_based_availability_strategy.py

```diff
@@ -71,7 +71,8 @@ def _check_list_files(self, stream: "AbstractFileBasedStream") -> List[RemoteFil
         try:
             files = stream.list_files()
         except Exception as exc:
-            raise CheckAvailabilityError(FileBasedSourceError.ERROR_LISTING_FILES, stream=stream.name) from exc
```

Comment:
I think what I'd do here is continue to raise the CustomException from your previous version, but catch it and re-raise with a CheckAvailabilityError, like this:

```
try:
    ...
except CustomFileBasedException as exc:
    raise CheckAvailabilityError(exc.error_message, stream=stream.name) from exc
except Exception as exc:
    ...
```

This way we can avoid having to do the `hasattr` check. Plus, I think we'll probably want to use your CustomFileBasedException in other places too as the file-based CDK matures.

Reply:
I've updated the code to use CustomFileBasedException and re-raise as CheckAvailabilityError, removing the hasattr check. Thanks for the suggestion.